src/components/contextual/pages/pools/StakedPoolsTable.vue

```diff
@@ -1,60 +1,77 @@
 <script setup lang="ts">
 import { computed } from 'vue';
 import { useI18n } from 'vue-i18n';
-
 import PoolsTable from '@/components/tables/PoolsTable/PoolsTable.vue';
-import useStaking from '@/composables/staking/useStaking';
 import { isL2 } from '@/composables/useNetwork';
 import { configService } from '@/services/config/config.service';
 import useWeb3 from '@/services/web3/useWeb3';
+import { useUserStaking } from '@/providers/local/user-staking.provider';
+import { Pool } from '@/services/pool/types';
+import { useUserPools } from '@/providers/local/user-pools.provider';
+import { sleep } from '@/lib/utils';
+import StakePreviewModal from '../pool/staking/StakePreviewModal.vue';
+import { providePoolStaking } from '@/providers/local/pool-staking.provider';
+
+/**
+ * STATE
+ */
+const showUnstakeModal = ref(false);
+const poolToUnstake = ref<Pool | undefined>();
+
+/**
+ * PROVIDERS
+ */
+providePoolStaking();
 
 /**
  * COMPOSABLES
  */
-const {
-  userData: {
-    stakedPools,
-    isLoadingUserStakingData,
-    isLoadingStakedPools,
-    isLoadingUserPools,
-    poolBoosts,
-  },
-} = useStaking();
+const { stakedPools, poolBoostsMap, stakedShares, isLoading } =
+  useUserStaking();
+const { refetchAllUserPools } = useUserPools();
 const { isWalletReady, isWalletConnecting } = useWeb3();
 const { t } = useI18n();
 const networkName = configService.network.shortName;
 
 /**
  * COMPUTED
  */
-const isLoading = computed((): boolean => {
-  return (
-    isLoadingUserStakingData.value ||
-    isLoadingStakedPools.value ||
-    isLoadingUserPools.value
-  );
-});
-
 const noPoolsLabel = computed(() => {
   return isWalletReady.value || isWalletConnecting.value
     ? t('noStakedInvestments', [networkName])
     : t('connectYourWallet');
 });
 
-const poolsWithBoost = computed(() => {
-  return stakedPools.value.map(pool => ({
-    ...pool,
-    boost: (poolBoosts.value || {})[pool.id],
-  }));
-});
-
 const hiddenColumns = computed(() => {
   const _hiddenColumns = ['poolVolume', 'migrate', 'lockEndDate'];
   if (isL2.value) _hiddenColumns.push('myBoost');
+
   return _hiddenColumns;
 });
 
-const poolsToRenderKey = computed(() => JSON.stringify(poolsWithBoost.value));
+const poolsToRenderKey = computed(() => JSON.stringify(stakedPools.value));
+
+/**
+ * METHODS
+ */
+function handleUnstake(pool: Pool) {
+  showUnstakeModal.value = true;
+  poolToUnstake.value = pool;
+}
+
+function handleModalClose() {
+  refetchAllUserPools();
+  showUnstakeModal.value = false;
+}
+
+async function handleStakeSuccess() {
+  await sleep(3000);
+  await refetchAllUserPools();
+}
+
+onMounted(() => {
+  refetchAllUserPools();
```

Comment:
why call this in onMounted and not just in setup function?

Reply:
Hmm the reason was to ensure the queries are refetched when a user returns to the portfolio page in case they've just done a stake/unstak on a pool page which would cause a pool to move from one table to the other. If refetch is not called then I think it can be the case where the query is not necessarily re-triggered and I'm not sure if I just have it in the setup function if it will be triggered if the user had previously been on the portfolio page?

Reply:
hm not sure here, should be tested. 
my point is just that if you put this call in setup function, it will occur a little bit earlier, so request will start before component is mounted